### PR TITLE
Rework numpy reader tests

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- Refactoring to improve variability in numpy reader tests. Variable shapes, batch sizes, num threads, etc.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Rewrote some of the NumpyReader tests to include a wider range of options*
 - Affected modules and functionalities:
     *NumpyReader tests*
 - Key points relevant for the review:
     *Is it covering all the important use cases?*
 - Validation and testing:
     *NA*
 - Documentation (including examples):
     *NA*

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

File: dali/test/python/test_operator_readers_numpy.py [hunk 1]
```diff
@@ -75,163 +87,108 @@ def NumpyReaderPipeline(path, batch_size, device="cpu", file_list=None, files=No
...
-            pipe.build()
-
-            for batch in range(0, num_samples, batch_size):
-                pipe_out = pipe.run()
-                arr_rd = pipe_out[0].as_array()
-                arr_np = arr_np_all[batch:batch+batch_size, ...]
+def test_types_and_shapes():
+    cache_header_information = False
+    for device in ["cpu", "gpu"] if is_gds_supported() else ["cpu"]:
+        for fortran_order in [False, True, None]:
```
Comment (JanuszL) [inline dali/test/python/test_operator_readers_numpy.py:158]: Maybe we can just go with None that means random, what do you think?

Comment (jantonguirao) [reply dali/test/python/test_operator_readers_numpy.py:158]: You mean, only test random?

**Review by JanuszL — APPROVED** [review]

File: dali/test/python/test_operator_readers_numpy.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_operator_readers_numpy.py:158]: Yes, only test random. Food for thoughts.

**Review by awolant — APPROVED** [review]

**Pull request merged**